internal/controller/etcdcluster_controller_test.go

```diff
@@ -30,55 +28,69 @@ import (
 	operatorv1alpha1 "go.etcd.io/etcd-operator/api/v1alpha1"
 )
 
-var _ = Describe("EtcdCluster Controller", func() {
-	Context("When reconciling a resource", func() {
-		const resourceName = "test-resource"
+func TestEtcdClusterController(t *testing.T) {
+	ctx := t.Context()
 
-		ctx := context.Background()
+	const resourceName = "test-etcd-cluster"
+	typeNamespacedName := types.NamespacedName{
+		Name:      resourceName,
+		Namespace: "etcd-operator-system",
+	}
 
-		typeNamespacedName := types.NamespacedName{
-			Name:      resourceName,
-			Namespace: "default", // TODO(user):Modify as needed
-		}
+	t.Run("When reconciling a resource", func(t *testing.T) {
 		etcdcluster := &operatorv1alpha1.EtcdCluster{}
 
-		BeforeEach(func() {
-			By("creating the custom resource for the Kind EtcdCluster")
-			err := k8sClient.Get(ctx, typeNamespacedName, etcdcluster)
-			if err != nil && errors.IsNotFound(err) {
-				resource := &operatorv1alpha1.EtcdCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceName,
-						Namespace: "default",
-					},
-					// TODO(user): Specify other spec details if needed.
-				}
-				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+		t.Log("Creating or retrieving the EtcdCluster resource")
+		err := k8sClient.Get(ctx, typeNamespacedName, etcdcluster)
+		if err != nil && errors.IsNotFound(err) {
+			resource := &operatorv1alpha1.EtcdCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      resourceName,
+					Namespace: "default",
+				},
 			}
-		})
+			if createErr := k8sClient.Create(ctx, resource); createErr != nil {
+				t.Fatalf("Failed to create EtcdCluster resource: %v", createErr)
+			}
+		} else if err != nil {
+			t.Fatalf("Failed to get EtcdCluster resource: %v", err)
+		}
 
-		AfterEach(func() {
-			// TODO(user): Cleanup logic after each test, like removing the resource instance.
+		// Defer a cleanup function to remove the resource after the test finishes.
+		defer func() {
```

Comment:
The deferred function get executed when the sub test complete. It means that the resource has already been removed before your next sub test "reconcile".

Pls just remove the sub test. ( I mean just remove `t.Run`, and keep the test code in the parent test function directly)